components/Checkbox.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+    <div class="checkbox-container">
+      <input type="checkbox" :checked="checked" :disabled="disabled" @change="onInputChange">
+      <label>{{ label }}</label>
+    </div>
+  </template>
+  
+  <script lang="ts">
+  import { defineComponent } from 'vue';
+  
+  export default defineComponent({
+    props: {
+      label: {
+        type: String,
+        required: true
+      },
+      checked: {
+        type: Boolean,
+        required: true
+      },
+      disabled: {
+        type: Boolean,
+        default: false
+      }
+    },
+    methods: {
+      onInputChange(event: Event) {
+        const target = event.target as HTMLInputElement;
+        this.$emit('update:checked', target.checked);
+      }
+    }
+  });
+  </script>
+  
+  <style scoped>
+  .checkbox-container input[type="checkbox"] {
+    border: none;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    appearance: none;
+    outline: none;
+    padding: 0;
+    margin: 0;
+    width: 1.2em;
+    height: 1.2em;
+    vertical-align: middle;
+    border: 1px solid #E7E7E7;
```

Comment:
Evite usar cores hardcoded com hex, todas elas já foram configuradas no tailwind. aqui o hex `#E7E7E7` representa `black-10`, logo, para aplicar a bordar, poderia ser:
```suggestion
    @apply border-black-10;
    border-width: 1px;
```